### ISSUE TITLE
Headless true produces different variables (unexpected results) in evaluate than Headless false (expected results)

### ISSUE THREAD

**Issue by Wkasel** [puppeteer/puppeteer#4886]
### Steps to reproduce

* 1.6.2 && 1.16.x (tried both)
* OSX 10.14.6:
* https://gist.github.com/Wkasel/fcd11589f37bd388a02e16430a113fbb
* Node.js version: 11.5.0

**What steps will reproduce the problem?**

Full code snippet in gist
```javascript
const puppeteer = require("puppeteer");

const urls = [
  `https://shop.nordstrom.com/c/womens-dresses-shop`,
  `https://shop.nordstrom.com/c/womens-dresses-shop?offset=10&page=2`
];

(async () => {
  try {
    const loadPage = async url => {
      /*
      // this is the problem area
      // when I have headless as false I get data for `page 2`
      // when I have headless true I get data for `page 1` when I expect `page 2`
      // for another test try this snippet in your browser
      // __INITIAL_CONFIG__.productResults.query.pageSelected
      // on `page 1` you should get 1, `page 2` you should get 2
      // the only exception is you must have a fresh load (don't just navigate to page 2)
      */
      const browser = await puppeteer.launch({ headless: true });
      // const browser = await puppeteer.launch({ headless: false });
      console.log(`testing with ${url}`);
      const page = await browser.newPage();
      await page.goto(url, {
        waitUntil: "networkidle0"
      });
      await page.waitFor(5000);
      /*
      // tried both of these solutions to test and it doesn't work
      //
      // await page.waitForSelector("#product-results-query-anchor");
      // await page.waitFor(5000);
      */

      const data = await page.evaluate(e => window.__INITIAL_CONFIG__);
      console.log(data.productResults.query.pageSelected);
      await page.close();
      await browser.close();
    };
    await loadPage(urls[0]);
    await loadPage(urls[1]);
  } catch (e) {
    console.log(e);
  }
})();
```

TL;DR; Comments says it all. Headless true produces different variables (unexpected results) in evaluate than Headless false (expected results)

side note: if you curl this page directly, `https://shop.nordstrom.com/c/womens-dresses-shop?offset=10&page=2` and look at the 
object `window.__INITIAL_CONFIG__.productResults.query.pageSelected` it stays as page 1, but eventually updates to page 2 when loaded in a browser.

**Comment by kblok** [comment]
I took a screenshot of the second page. And you are getting a popup on headless false.
You might need to close that popup and then redirect to the second page. I got many different countries while getting to that page...

![pager](https://user-images.githubusercontent.com/2198466/63936389-1754f800-ca36-11e9-99fb-7cd29ce5d4ed.png)
